src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -200,7 +210,7 @@ public void UpdateTrackedDocument(Uri uri, SourceText newSourceText)
 
         // Add the document to our loose files workspace if its open.
         var miscDocument = _trackedDocuments.ContainsKey(uri) ? _lspMiscellaneousFilesWorkspace?.AddMiscellaneousDocument(uri, _trackedDocuments[uri], _logger) : null;
```

Comment:
the code here to add the document to miscellaneous files should only run for the `Document` version of `GetLspDocumentAsync` - thats because the misc files workspace only supports real documents at the moment.  So I think these two lines just need to move into that method